### PR TITLE
feat: Name & WriteLowPri methods

### PR DESCRIPTION
**Pull request by iproudhon**
Added Name() for easier debugging & and WriteLowPri() for pruning.

### CODE REVIEW

File: rdb/batch.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+func (b *rdbBatch) Write() error {
+	if b.b == nil {
+		return tmdb.ErrBatchClosed
+	}
+	var cerr *C.char
+	C.rocksdb_write(b.db.db, b.db.wopts, b.b, &cerr)
+	if cerr != nil {
+		return cerror(cerr)
```
Comment (zemyblue) [inline rdb/batch.go:59]: No need to close the `d` as `rdbBatch` in this case?

File: rdb/batch.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+func (b *rdbBatch) WriteSync() error {
+	if b.b == nil {
+		return tmdb.ErrBatchClosed
+	}
+	var cerr *C.char
+	C.rocksdb_write(b.db.db, b.db.wsopts, b.b, &cerr)
+	if cerr != nil {
+		return cerror(cerr)
```
Comment (zemyblue) [inline rdb/batch.go:72]: ditto.

File: rdb/batch.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+func (b *rdbBatch) WriteLowPri() error {
+	if b.b == nil {
+		return tmdb.ErrBatchClosed
+	}
+	var cerr *C.char
+	C.rocksdb_write(b.db.db, b.db.wlpopts, b.b, &cerr)
+	if cerr != nil {
+		return cerror(cerr)
```
Comment (zemyblue) [inline rdb/batch.go:85]: same as above

File: rdb/db.go [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+	C.rocksdb_block_based_options_set_block_cache(bbto, C.rocksdb_cache_create_lru(C.size_t(1<<30)))
+	C.rocksdb_block_based_options_set_filter_policy(bbto, C.rocksdb_filterpolicy_create_bloom(C.int(10)))
+	defer C.rocksdb_block_based_options_destroy(bbto)
+
+	opts := C.rocksdb_options_create()
+	C.rocksdb_options_set_block_based_table_factory(opts, bbto)
+	C.rocksdb_options_set_create_if_missing(opts, 1)
+	C.rocksdb_options_increase_parallelism(opts, C.int(runtime.NumCPU()))
+	C.rocksdb_options_optimize_level_style_compaction(opts, 512*1024*1024)
+	C.rocksdb_options_optimize_level_style_compaction(opts, 512*1024*1024)
```
Comment (zemyblue) [inline rdb/db.go:72]: Why is this line executed twice?

File: rdb/db.go [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+	C.rocksdb_options_set_enable_pipelined_write(opts, 1)
+
+	ropts := C.rocksdb_readoptions_create()
+	wopts := C.rocksdb_writeoptions_create()
+	wsopts := C.rocksdb_writeoptions_create()
```
Comment (zemyblue) [inline rdb/db.go:77]: I don't know what each opts does, could you add some comment if possible?

File: rdb/db.go [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+	return db.name
+}
+
+func (db *RDB) Get(key []byte) ([]byte, error) {
+	var cerr *C.char
+	var cvl C.size_t
+	ck := b2c(key)
+	cv := C.rocksdb_get(db.db, db.ropts, ck, C.size_t(len(key)), &cvl, &cerr)
+	if cerr != nil {
+		err := cerror(cerr)
```
Comment (zemyblue) [inline rdb/db.go:112]: I think to need to free `unsafe.Pointer(cv)` before return. How about using `defer` of golang?

File: makefile [hunk 7]
```diff
@@ -101,6 +111,27 @@ format:
...
+cleveldb:
+	@if [ ! -e $(CLEVELDB_DIR) ]; then         \
+		sh contrib/get_cleveldb.sh;        \
+	fi
+	@if [ ! -e $(CLEVELDB_DIR)/libcleveldb.a ]; then   \
+		cd $(CLEVELDB_DIR);                        \
+		mkdir build;                               \
+		cd build;                                  \
+		cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DLEVELDB_BUILD_TESTS=OFF -DLEVELDB_BUILD_BENCHMARKS=OFF ..; \
+		make;                                      \
```
Comment (0Tech) [inline makefile:124]: Let's do it CMake way.
```suggestion
		cmake -S . -B build -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DLEVELDB_BUILD_TESTS=OFF -DLEVELDB_BUILD_BENCHMARKS=OFF; \
		cmake --build build;                                      \
```

File: rdb/db.go (hunk 6 above)
Comment (dudong2) [reply rdb/db.go:112]: I think cv is nil if rocksdb_get return an error. So, maybe free(unsafe.Pointer(cv)) isn't needed in this case.

Comment (dudong2) [reply rdb/db.go:112]: I think cv is nil if rocksdb_get return an error. So, maybe free(unsafe.Pointer(cv)) isn't needed in this case.

**Review by 0Tech — CHANGES_REQUESTED** [review]
Not critical, but some improvements.

File: rdb/batch.go (hunk 1 above)
Comment (iproudhon) [reply rdb/batch.go:59]: agreed.

Comment (iproudhon) [reply rdb/batch.go:60]: Moved b.Close up.

File: rdb/batch.go (hunk 2 above)
Comment (iproudhon) [reply rdb/batch.go:73]: Moved b.Close up.

File: rdb/batch.go (hunk 3 above)
Comment (iproudhon) [reply rdb/batch.go:86]: Moved b.Close up.

File: rdb/db.go (hunk 5 above)
Comment (iproudhon) [reply rdb/db.go:76]: Added comments inside struct definition.

File: rdb/db.go (hunk 4 above)
Comment (iproudhon) [reply rdb/db.go]: removed it

**Review by dudong2 — APPROVED** [review]

**Review by 0Tech — APPROVED** [review]
LGTM

File: rdb/iterator.go [hunk 8]
```diff
@@ -0,0 +1,98 @@
+//go:build rocksdb
+// +build rocksdb
+
+package rdb
+
+// #include <stdlib.h>
+// #include "rocksdb/c.h"
```
Comment (zemyblue) [inline rdb/iterator.go:7]: Please remove this line, if don't need any more.

**Pull request merged**